### DISCUSSION TITLE
Playtesting Iris

### DISCUSSION THREAD

**Discussion by FoundationGames** [IrisShaders/Iris#67] [General]
# I want to playtest this mod! Where's the download?

## Long Answer:
If you are **experienced with compiling mods from source code**, you may pull Iris from source and run the mod that way.
### Why?
**Iris is WIP.** Not all features are actually implemented yet.
When users who are not experienced with using such WIP software are given easy access to said software, it creates a support burden on the maintainers of the project. It may cause users to flood discussion areas asking for help with making the mod work properly, and that burdens the maintainers with needing to help all these people.
A majority of the time, providing these users help will not create much benefit for the maintainers either, as many will not submit adequate/helpful bug reports or submit pull requests to the mod. So all that effort ends up not being worth it.
#### If you have compiled the mod and want to know how to use it:
You may drag shader packs into the `.minecraft/shaderpacks/` (not necessarily `.minecraft/`, can be whatever game instance folder you are using) or into the shader pack selection UI. <br/>
The shader pack selection UI is accessible in `Options > Video Settings > Shader Packs` **if you are not using Sodium** (the iris compatible branch of Sodium). If you are using Sodium, you will need to press `Shift+P` in the Sodium video settings screen in order to access the vanilla screen. A better solution for this is planned. <br/>
Once in the shader pack selection UI, click on the pack you would like to use. Either click Apply (to save changes without exiting the screen) or Done (saves and exits) to apply the shader pack. Clicking Cancel will discard your changes, if there are any that are unsaved.

**Comment by BeansBeefBroccoli** [comment]


**Reply by ThatCreeper** [reply]
What did it say in the console?

**Reply by FoundationGames** [reply]
Make sure that you have a proper JDK installed, and you ran `gradlew build` with the end message saying `BUILD SUCCESSFUL`. If you have any further questions, join the Iris discord.

**Reply by BeansBeefBroccoli** [reply]
I'll try to check if I have the correct JDK installed, and do it again to see if that works, it did say BUILD SUCCESSFUL when it was done.

**Reply by BeansBeefBroccoli** [reply]
Nevermind, I did it the way OverlordsIII said and I have the Jar file now.

**Comment by OverlordsIII** [comment]
You can use github actions at the top of the screen to get the jar without compiling it, but then you wouldn't be able to make quick changes to the code

**Reply by FoundationGames** [reply]
Added this to the post, thanks

**Comment by Moosey-Moto** [comment]
Hey, I've tried running both methods, and even tried the sodium compatible branch just because. Every time I get the mod to load, but don't have any shaders show up. I can hit R and get a shaders reloaded message but everything looks like default MC. Am I missing something super obvious? AMD cpu and gpu if that makes a difference. Thanks!

**Reply by IMS212** [reply]
You need to put a shader in the shaderpacks folder, and select it using config\iris.properties by replacing (internal) with the shaderpack file name.

**Reply by Moosey-Moto** [reply]
Tried multiple shaders. Even Internal just comes up with something that looks like default

**Reply by Moosey-Moto** [reply]
Internal, Sildurs, SEUS, and continuum all look the same. Probably still coming up with the internal one. I'd happily send you a screenshot of my config if you want it.

**Reply by FoundationGames** [reply]
There are some known issues with AMD drivers, however these issues involve crashing. You should probably report your problems as an Issue and detail your hardware configuration.

**Reply by Moosey-Moto** [reply]
Figured out I was being dumb this morning after some sleep. Now just having the known crashing issues. Thanks guys, and sorry!

**Reply by lawrencfgsdfg** [reply]
how'd you fix it

**Comment by DARK0717** [comment]
Im trying to build Iris for 21w06a, snapshot of 1.17 to get shaders working, i have seen someone do this but I am now facing 29 errors saying "cannot find symbol" in uniforms, shaderpipeline, celestial uniforms, mixin, (class Option, class Vector4f etc) etc

**Reply by OverlordsIII** [reply]
Yes atm Iris does not support snapshots and some new things have to be done to be able to run it on the new snapshots.

**Comment by SkyTheNerd** [comment]


**Reply by CrazycatASG-zz** [reply]
did you make sure to change to your iris folder in command line? because that is what gradle is saying

**Comment by IMS212** [comment]
lol
